number_theory/prime.h

```diff
@@ -50,6 +51,9 @@ class Sieve {
   Sieve &operator=(const Sieve &) = default;
   Sieve &operator=(Sieve &&) = default;
 
+  // Returns the maximum number (inclusive) we can hold.
+  const T &get_limit() const { return num_limit_; }
```

Comment:
This made me notice that our `is_prime` function is not const. Would you mind making `is_prime` const as well?